### PR TITLE
fix(unit tests): Panic caused by edge case during unit test build

### PR DESCRIPTION
**Pull request by Kalopsian-tz**
## Summary

I was working on the unit test included in the config documented below which is **invalid** because `filter_msg` is listed in both `no_outputs_from` and `outputs.extract_from`.

The issue arises when you try to run the invalid test, instead of handling the issue cleanly, Vector panics instead:
```
$ vector test vector.yaml
thread 'main' panicked at src/config/unit_test/mod.rs:75:49:
An unexpected error occurred while executing unit tests. Please try again.: RecvError(())
```

This seems to happen because Vector doesn't include a check that `output.extract_from` entries are not also listed in `no_outputs_from` which allows it to pass the build phase and fail when the invalid test is unwrapped, causing the panic.

This PR adds a check which resolves the issue and lets Vector exit cleanly with a defined error that follows the format used by another error in the same function.
```
$ vector test vector.yaml
Running tests
2025-08-22T01:59:08.590667Z ERROR vector::unit_test: Failed to execute tests:
Failed to build test 'Invalid test':
  Invalid extract_from target in test 'Invalid test': 'filter_msg' listed in no_outputs_from.
```

PS: This is my first time raising a PR on an open source project so any feedback on how I raised / worded this PR would be greatly appreciated! (I hope its ok I made a PR without making an issue first? The contributing page said "For large PRs or for breaking changes, ensure your change has an issue!" I couldn't find another issue raised about this and figured this wasn't a large or breaking change)

## Vector configuration

```yaml
sources:
  cmd:
    type: exec
    command:
      - echo
      - "a message!"
    mode: scheduled
    scheduled:
      exec_interval_secs: 120

transforms:
  filter_msg:
    type: filter
    inputs:
      - cmd
    condition: |
      .message != "a message!"

tests:
- name: Invalid test
  no_outputs_from:
    - filter_msg
  inputs:
  - insert_at: filter_msg
    type: log
    log_fields:
      message: "a message!"
  outputs:
  - extract_from: filter_msg
    conditions:
    - type: vrl
      source: |
        assert_eq!(.message, "a message!")

sinks:
  console:
    type: console
    encoding:
      codec: json
    inputs:
      - filter_msg
```

## How did you test this PR?

I trialed similar edge cases I could think of to ensure they also pass with this new check as well as running the suggested tests:
```
cargo fmt --all
cargo clippy --workspace --all-targets -- -D warnings
```

## Change Type
- [x] Bug fix
- [ ] New feature
- [ ] Non-functional (chore, refactoring, docs)
- [ ] Performance

## Is this a breaking change?
- [ ] Yes
- [x] No

## Does this PR include user facing changes?

- [x] Yes. Please add a changelog fragment based on our [guidelines](https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md).
- [ ] No. A maintainer will apply the `no-changelog` label to this PR.

## Notes
- Please read our [Vector contributor resources](https://github.com/vectordotdev/vector/tree/master/docs#getting-started).
- Do not hesitate to use `@vectordotdev/vector` to reach out to us regarding this PR.
- Some CI checks run only after we manually approve them.
  - We recommend adding a `pre-push` hook, please see [this template](https://github.com/vectordotdev/vector/blob/master/CONTRIBUTING.md#Pre-push).
  - Alternatively, we recommend running the following locally before pushing to the remote branch:
    - `cargo fmt --all`
    - `cargo clippy --workspace --all-targets -- -D warnings`
    - `cargo nextest run --workspace` (alternatively, you can run `cargo test --all`)
- After a review is requested, please avoid force pushes to help us review incrementally.
  - Feel free to push as many commits as you want. They will be squashed into one before merging.
  - For example, you can run `git merge origin master` and `git push`.
- If this PR introduces changes Vector dependencies (modifies `Cargo.lock`), please
  run `cargo vdev build licenses` to regenerate the [license inventory](https://github.com/vectordotdev/vrl/blob/main/LICENSE-3rdparty.csv) and commit the changes (if any). More details [here](https://crates.io/crates/dd-rust-license-tool).

### CODE REVIEW

File: changelog.d/panic_from_invalid_unit_test_build.fix.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Fixed panic during unit test when a `transform` is listed in both `no_outputs_from` and `output.extract_from`.
```
Comment (pront) [inline changelog.d/panic_from_invalid_unit_test_build.fix.md]: Hi @Kalopsian-tz, I think we can simplify this and phrase it from a user standpoint.

**Review by pront — APPROVED** [review]

File: changelog.d/panic_from_invalid_unit_test_build.fix.md (hunk 1 above)
Comment (Kalopsian-tz) [reply changelog.d/panic_from_invalid_unit_test_build.fix.md]: Hi @pront, would something more along the lines of this suit better?
"Added validation to ensure a test that expects no output from a source, does not perform operations on said source."

Comment (pront) [reply changelog.d/panic_from_invalid_unit_test_build.fix.md]: Yes 👍
